packages/compass-data-modeling/src/components/diagram-editor.tsx

```diff
@@ -166,14 +174,56 @@ const DiagramEditor: React.FunctionComponent<{
           };
           break;
         default:
-          throw new Error(`Unknown placeholder ${placeholder}`);
+          throw new Error(`Unknown placeholder ${type}`);
       }
       setApplyInput(JSON.stringify(placeholder, null, 2));
     };
 
-  const modelStr = useMemo(() => {
-    return JSON.stringify(model, null, 2);
-  }, [model]);
+  const edges = useMemo(() => {
+    return (model?.relationships ?? []).map((relationship): EdgeProps => {
+      const [source, target] = relationship.relationship;
+      return {
+        id: relationship.id,
+        source: source.ns,
+        target: target.ns,
+        markerStart: 'one',
+        markerEnd: 'many',
```

Comment:
Is this a placeholder item?

Something that's become fairly clear over time is that we're likely going to want to work with concrete cardinalities if we can (i.e. number ranges rather than `one` or `many`)